### ISSUE TITLE
Provide the widget preview iframe with a sidebar ID

### ISSUE THREAD

**Issue by dikiyforester** [WordPress/gutenberg#33638]
## What problem does this address?
There is no way for a block/widget to identify and get data on the sidebar it’s been placed on. 

Prior widget editor, we could get the sidebar id by the widget id, but in block-based widget editor it's impossible since widget id in the editor is not the actual widget id.

For example:
1. Some would like to get the sidebar’s before_title and after_title variables from within the block itself.
2. Some would like to add extra widget fields, depending on the sidebar.

Reported here: 
1. https://make.wordpress.org/core/2021/06/29/block-based-widgets-editor-in-wordpress-5-8/#comment-41612
2. https://make.wordpress.org/core/2021/06/29/block-based-widgets-editor-in-wordpress-5-8/#comment-41618

## What is your proposed solution?
Add the current sidebar id variable to the block/widget instance in block-based widgets editor.

**Comment by kevin940726** [comment]
If I'm not misunderstanding this, I think it's a duplicate of #32805?

**Comment by cabrerahector** [comment]
Not really, @kevin940726. What we want is for the block to be able to access its sidebar data (id, before_widget, after_widget, before_title, after_title, etc).

The issue you're referring to requests for blocks to get assigned a widget_id on the Widgets screen which is sort of similar but not exactly the same thing.

**Comment by kevin940726** [comment]
@cabrerahector See https://github.com/WordPress/gutenberg/issues/32805#issuecomment-880376265, I think we ultimately want the same thing for preview iframes to have the sidebar ids. Am I missing something obvious? 🤔

**Comment by noisysocks** [comment]
We can’t provide the preview iframe with a widget ID because the ID might not exist in all cases. For example, if a user creates a new Legacy Widget, the ID isn’t created until the user presses _Update_.

Potentially, we could pass `sidebar_id` [as a widget argument](https://github.com/WordPress/gutenberg/blob/d3688d8d6f1388a6805c62bd2a51556ba541f99b/packages/widgets/src/blocks/legacy-widget/index.php#L45-L48) so that plugins can access `$args['sidebar_id']` in `WP_Widget::widget()` to work around this limitation.
